

# Q1 Number and string format
1. Assign a String variable s to a greeting statement. Then print s.
2. Assign a Character variable a to a character. Then print a.
3. Assign an Integer variable x to an integer number. Then print x.
4. Assign a Double variable y to a Double number. Then print y.
5. Assign a Float variable z to a float number. Then print z.
6. Assign an Integer variable x Then print x.

In [1]:
# 1.1
s = "Hi Class"
print(s)


Hi Class


In [2]:
# 1.2
a = chr(97)
print(a)

a


In [3]:
# 1.3
x = 1
print(x)

1


In [7]:
# 1.4
y = 10.00
print(y)

10.0


In [6]:
# 1.5
z = 1.0
print(z)

1.0


In [8]:
# 1.6
x = 4
print(x)


4


# Q2 Manipulating Numbers
Given an integer, a=123456789
1. Print the binary number for a
2. Print the the integer ax10,ax100, a+100, a%10, a%100, a÷10, a÷100
3. Print the binary number a after shifting by 1, 2, and 4 to the left. Print the numbers in
decimal. Explain the result.
4. Print the binary number a after shifting by 1, 2 and 4 to the right. Print the numbers in
decimal. Explain the result.

In [9]:
# 2.1
a=123456789
a_bin = bin(a)

print(a_bin)

0b111010110111100110100010101


In [12]:
# 2.2
print(a*10)
print(a*100)
print(a+100)
print(a%10)
print(a%100)
print(int(a/10))
print(int(a/100))


1234567890
12345678900
123456889
9
89
12345678
1234567


In [19]:
# 2.3
print(a<<1)
print(a<<2)
print(a<<4)

246913578
493827156
1975308624


In [23]:
print(int('111010110111100110100010101', 2))
print(int('1110101101111001101000101010', 2))
print(int('11101011011110011010001010100', 2))
print(int('1110101101111001101000101010000', 2))


123456789
246913578
493827156
1975308624


### Explanation
<br>
a_bin = 111010110111100110100010101
<br>
shift 1 to the left = 1110101101111001101000101010, which is 246913578 in decimal
<br>
shift 2 to the left = 11101011011110011010001010100, which is 493827156 in decimal
<br>
shift 4 to the left = 1110101101111001101000101010000, which is 1975308624 in decimal

In [24]:
# 2.4
print(a >> 1)
print(a >> 2)
print(a >> 4)


61728394
30864197
7716049


In [25]:
print(int('111010110111100110100010101', 2))
print(int('11101011011110011010001010', 2))
print(int('1110101101111001101000101', 2))
print(int('11101011011110011010001', 2))


123456789
61728394
30864197
7716049


### Explanation
<br>
a_bin = 111010110111100110100010101
<br>
shift 1 to the right = 11101011011110011010001010, which is 61728394 in decimal
<br>
shift 2 to the right = 1110101101111001101000101, which is 30864197 in decimal
<br>
shift 4 to the right = 11101011011110011010001, which is 7716049 in decimal

# Q3 Manipulating Strings
Given a string s=”assalamu’alaikum”
1. Print the string s.
2. Print the first character in s.
3. Print the second character in s.
4. Print the last character in s.
5. Print the second last character in s.
6. Print the characters one to three in s.
7. Print the character 5 to the last in s. >>lamu'alaikum
8. Print the string s backwards. >> mukiala'umalassa
9. Print the the string s excluding the last 3 characters in s and add the string
“ka”
10. Print a new string by adding the string “Ahlan” to the last 8 characters in s.

In [27]:
# 3.1
s="assalamu’alaikum"
print(s)

assalamu’alaikum


In [38]:
# 3.2 first char
print(s[0])
# 3.3 second char
print(s[1]) 
# 3.4 last char
print(s[-1])
# 3.5 second last char
print(s[-2])
# 3.6 char one to three
print(s[0:3])
# 3.7 char 5 to the last
print(s[4:])
# 3.8 backwards
print(s[::-1])
# 3.9 exclude last 3 strings and add "ka"
print(s[:-3]+"ka")
# 3.10 
print("ahlan"+s[-8:])

a
s
m
u
ass
lamu’alaikum
mukiala’umalassa
assalamu’alaika
ahlan’alaikum


# Q4 DECOMPOSITION: Array/ List as data structure
<br>
Given an array mylist = [1, 2, 3, 4, 5, 6]
<br>

1. Print the length of the array
2. Print the first element in the array
3. Change the second element in the array to number 10, then print the whole array
4. Print each element in the array using a for loop
5. Print each element in the array using a while loop
6. Shift the array element to the right while the last element moved to the first position
7. Shift the array element to the left while the first element moved to the last position